### PR TITLE
DMP-2302: Admin portal - Edit group details

### PR DESCRIPTION
**Pull request by deanbullock-hmcts**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/DMP-2302

### Change description ###

- Can now edit group details
- Add group form for updating / creating groups
- Validation prevents naming a group the same as an existing group's name
- Role can only be added on creation and not on update
- Moved security groups interface/types to admin/models/groups
- Partially implemented create group, to be implemented in another ticket

### CODE REVIEW

File: src/app/admin/components/groups/create-edit-group/create-edit-group.component.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }
+  }
+
+  onSave(formValues: GroupFormValue) {
+    if (this.isEdit) {
+      this.groupService.updateGroup(this.group.id, formValues).subscribe(() => {
+        this.router.navigate(['admin/groups', this.group.id], { queryParams: { updated: true } });
+      });
+    }
+    // } else {
```
Comment (DavidTMann) [inline src/app/admin/components/groups/create-edit-group/create-edit-group.component.ts:44]: Is this supposed to be commented out?

Comment (deanbullock-hmcts) [reply src/app/admin/components/groups/create-edit-group/create-edit-group.component.ts:44]: yes - it's a gift for whoever picks up create group

Comment (DavidTMann) [reply src/app/admin/components/groups/create-edit-group/create-edit-group.component.ts:44]: 🤗

**Review by DavidTMann — APPROVED** [review]

**Review by bpires2 — APPROVED** [review]
💯

**Pull request merged**